daml/Marketplace/Distribution/Auction/Service.daml

```diff
@@ -120,10 +121,10 @@ template Service
 
           -- Split bids on those above / below the floor price -> Sort valid bids descendingly -> Execute Auction -> Split auction result on allocated / unallocated (ie, zero allocation)
           let
-            (validBids, invalidBids) = splitList (\bid -> bid.details.price < floorPrice) bids
+            (invalidBids, validBids) = partitionEithers $ splitList (\bid -> bid.details.price >= floorPrice) bids
```

Comment:
Why doesn't `goupBy` work? I don't think you need `partitionEithers`, as the left/right are the same type `a`. Something like:
```suggestion
            [invalidBids, validBids] = DA.List.groupBy (.details.price >= floorPrice) bids
```

Reply:
I commented on the chat on Friday why `groupBy` isn't ideal here - 

Hey @Luciano, I was looking at your comment on the `splitList` function -> I tried using `groupBy` but this isn’t what I’m looking for as we don't want to compare a bid to its adjacent bid in a list (it takes the function `(a -> a -> bool)`. `groupOn` looked to be what I wanted but when all elements either pass or fail the predicate (same issue with `groupBy`) => I have no easy way to know if all the elements in the list passed or failed the predicate without having to iterating again over the result.

Reply:
My bad, I think `groupOn` is what I wanted
```
luciano@DA-Luciano:~/src/da-marketplace$ daml repl
daml> import DA.List
daml> groupOn (>4) [1..10]
[[1,2,3,4],[5,6,7,8,9,10]]
daml> 
```

Reply:
Plus, you'll need a `non-exhaustive` pattern when changing the code to that suggest